x-pack/plugins/reporting/server/lib/tasks/execute_report.ts

```diff
@@ -324,12 +324,22 @@ export class ExecuteReportTask implements ReportingTask {
           this.logger.debug(`Reports running: ${this.reporting.countConcurrentReports()}.`);
 
           try {
-            const stream = await getContentStream(this.reporting, {
-              id: report._id,
-              index: report._index!,
-              if_primary_term: report._primary_term,
-              if_seq_no: report._seq_no,
-            });
+            const exportTypesRegistry = await this.reporting.getExportTypesRegistry();
+            const { jobContentEncoding } = exportTypesRegistry.get(
+              ({ jobType: item }) => item === jobType
+            );
```

Comment:
I think we should add a private member to this class that maps the jobTypeEncoding for each export type, and initialize the map in the init function's `for` loop. Similar to what is done with `this.taskExecutors`

Also, `reporting.getExportTypesRegistry` is synchronous - doesn't need await :)